### PR TITLE
Update Community page with projects and orgs built on Envoy Proxy

### PR DESCRIPTION
**Pull request by caniszczyk**
Envoy has a dynamic community of projects built on it and companies offering products and support. This is built upon the work of @trjordan https://github.com/envoyproxy/envoyproxy.github.io/pull/31

We propose listing the projects and orgs in accordance to the CNCF Website Guidelines:
https://www.cncf.io/projects/website-guidelines/

### CODE REVIEW

**Comment by caniszczyk** [comment]
Hey @mattklein123 and @richarddli let me know if you see any issues with this.

I've added a handful of projects/orgs to start so it should be good enough to merge. We can make a formal call on the users/dev mailing list for other folks to add them.

File: _data/home.yml [hunk 1]
```diff
@@ -90,3 +90,43 @@ quote:
...
+ - name: Ambassador
+   logo: /img/logos/ambassador.svg
+   website: https://www.getambassador.io/
+   github: https://github.com/datawire/ambassador
+   description: Ambassador is an open source Kubernetes-native API Gateway built on Envoy, designed for microservices.
+ - name: Cilium
+   logo: /img/logos/cilium.svg
+   website: https://github.com/cilium/cilium
+   github: https://github.com/cilium/cilium
+   description: Cilium is open source software for providing and transparently securing network connectivity and loadbalancing between application workloads such as application containers or processes. See https://github.com/cilium/cilium/issues/418 for Envoy support.
```
Comment (richarddli) [inline _data/home.yml]: I think we can delete the link to the GitHub issue? From reading it, it looks like they've integrated Envoy support already? cc: @tgraf

Comment (tgraf) [reply _data/home.yml]: Reference to GH issue can be removed. Envoy has been the default proxy for a while.

Comment (caniszczyk) [reply _data/home.yml]: k will fix! was more concerned about the project description which I cargo culted :)

**Pull request merged**